### PR TITLE
4.0 clause foreach improvement

### PR DESCRIPTION
**Pull request by martin-neotech**

### CODE REVIEW

**Comment by martin-neotech** [comment]
The result from the `query()` function needs to be fixed.

```
MATCH p=(start)-[*]->(finish)
WHERE start.name = 'A' AND finish.name = 'D'
FOREACH (n IN nodes(p) | SET n.marked = true)
```
       
This is how it is being formatted:

```
MATCH p=(
START)-[*]->(finish)
WHERE
START .name = 'A' AND finish.name = 'D'
FOREACH (n IN nodes(p)| SET n.marked = TRUE )
```

**Comment by martin-neotech** [comment]
This is the important change:

Within the `FOREACH` parentheses, you can do any of the updating commands -- `SET`, `REMOVE`, `CREATE`, `MERGE`, `DELETE`, and `FOREACH`.

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/ForeachTest.scala [hunk 1]
```diff
@@ -28,36 +28,28 @@ class ForeachTest extends DocumentingTest with QueryStatisticsTestSupport {
...
-      p("""If you want to execute an additional `MATCH` for each element in a list then `UNWIND` (see <<query-unwind>>) would be a more appropriate command.""".stripMargin)
+          #This means that if you `CREATE` a node variable within a `FOREACH`, you will _not_ be able to use it outside of the foreach statement, unless you match to find it.""".stripMargin('#'))
+      p("Within the `FOREACH` parentheses, you can do any of the updating commands -- `SET`, `REMOVE`, `CREATE`, `MERGE`, `DELETE`, and `FOREACH`.")
+      p("If you want to execute an additional `MATCH` for each element in a list then the <<query-unwind,`UNWIND`>> clause would be a more appropriate command.")
       graphViz()
     }
     section("Mark all nodes along a path", "foreach-mark-all-nodes-along-a-path") {
-      p(
-        """This query will set the property `marked` to true on all nodes along a path.""".stripMargin)
+      p("This query will set the property `marked` to true on all nodes along a path.")
```
Comment (AlexicaWright) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/ForeachTest.scala]:
```suggestion
      p("This query will set the property `marked` to `TRUE` on all nodes along a path.")
```

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/ForeachTest.scala [hunk 2]
```diff
@@ -28,36 +28,28 @@ class ForeachTest extends DocumentingTest with QueryStatisticsTestSupport {
...
       query(
-        """MATCH p = (begin)-[*]->(end)
-          |WHERE begin.name = 'A'
-          |AND end.name = 'D'
-          |FOREACH (n IN nodes(p) | SET n.marked = true)""".stripMargin, ResultAssertions((r) => {
+        """MATCH p=(start)-[*]->(finish)
```
Comment (AlexicaWright) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/ForeachTest.scala:50]: This renders with a line break after the _p_ in `MATCH p=(start)-[*]->(finish)`

**Review by AlexicaWright — COMMENTED** [review]
Looks great, Thank you @martin-neotech , just two comments. Good riddance with the bullet points!

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/ForeachTest.scala (hunk 2 above)
Comment (martin-neotech) [reply cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/ForeachTest.scala:50]: Yes i will try to fix the uglyfier in a nother PR.

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/ForeachTest.scala (hunk 1 above)
Comment (martin-neotech) [reply cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/ForeachTest.scala]: Write boolean literals (true and false) in lower case. - https://neo4j.com/docs/cypher-manual/4.2/styleguide/

Comment (martin-neotech) [reply cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/ForeachTest.scala]: so i will fix it to be `true`

**Pull request merged**